### DISCUSSION TITLE
pac data - allow setting number of input threads

### DISCUSSION THREAD

**Discussion by martintoelk** [microsoft/powerplatform-vscode#552] [Ideas]
Hi PAC CLI Team,

the CMT has an undocumenten parameter called "numberofinputthreads". In past, when you downloaded the CMT it was set to 5. In PAC CLI, it seems that this param is set to 1, which highly slow down the import speed. 

It would be great, if we can specifiy the numberofinputthreads param via pac cli. for example pac data - import --numberofinputthreads 20 -d "importpath"

numberofinputthreads set to 19 in app.config
![image](https://user-images.githubusercontent.com/24940711/235135771-cc63b3f3-24b9-4676-b078-ca3ddcbfb415.png)

CMT Desktop Verison will use 19 Threads
![image](https://user-images.githubusercontent.com/24940711/235136043-e1c40357-af30-48a8-a19d-3270b8b0d982.png)

pac cli data import only uses 1 Thread in default configuration
![image](https://user-images.githubusercontent.com/24940711/235136388-8fcb3526-96d3-4f77-980c-4c13bc5a2249.png)

**Comment by MailboxMember** [comment]
I was also curious about this setting yesterday, and couldn't find any documentation for it as well...
Does it effect the speed of the import when you where using CMT with 19 instead of 5? 
Will it have any effect on the Export as well? 
Did you try changing the setting for the PAC CLI?

**Reply by martintoelk** [reply]
**_Does it effect the speed of the import when you where using CMT with 19 instead of 5?_**
Seems so. Did a short ugly test, with a small Dataset. Result might be even better in reality. I am using home wifi behind VPN 
![image](https://github.com/microsoft/powerplatform-vscode/assets/24940711/92805600-18c2-4657-84ee-e9f12a27c6b3)

**_Will it have any effect on the Export as well?_**
Dont tested it because export does not take much time.

**_Did you try changing the setting for the PAC CLI?_**
Cant check it, company Proxy is biting me

**Reply by rajyraman** [reply]
@petrochuk - If there is any way to increase the perf in pipeline it would be great. Too slow at the moment. In the pipeline logs I am mostly seeing "threaded (1)"

![image](https://github.com/microsoft/powerplatform-vscode/assets/5035266/5ab744ad-c6d2-44a9-968d-2fa3424ccf54)
